tasks/undercloud.yml

```diff
@@ -0,0 +1,61 @@
+- block:
+    - name: create undercloud.conf
+      template:
+        src: undercloud.conf
+        dest: /home/stack/undercloud.conf
+        owner: stack
+        group: stack
+        mode: 0644
+
+    - name: install packages
+      package:
+        name: "{{ openstack_undercloud_packages }}"
+        state: present
+      register: _undercloud_pkg_install
+      retries: 3
+      until: _undercloud_pkg_install is success
+
+    - name: clean yum - workaround stale data
+      command: yum clean all
+
+    - name: yum repolist - workaround stale data
+      command: yum repolist
+
+    - name: create hiera_override.yaml
+      copy:
+        content: "nova::compute::ironic::max_concurrent_builds: 5"
+        dest: /home/stack/hiera_override.yaml
+        owner: stack
+        group: stack
+        mode: 0644
+  become: "{{ openstack_undercloud_become }}"
+  become_user: "{{ openstack_undercloud_become_user }}"
+
+- name: install undercloud
+  become: true
+  become_user: stack
+  command: openstack undercloud install
+  changed_when: false
+  register: _install_undercloud
+  retries: 2
+  until: _install_undercloud is success
+
+- name: list running openstack services
+  become: "{{ openstack_undercloud_become }}"
+  become_user: "{{ openstack_undercloud_become_user }}"
+  command: systemctl list-units openstack-*
+  register: _list_openstack_services
+  changed_when: false
+
+- name: display running openstack services
+  debug:
+    var: _list_openstack_services.stdout
+
+- name: append CA bundles path to stackrc
+  become: true
+  become_user: stack
+  lineinfile:
```

Comment:
Can `blockinfile` be used here? I prefer it for cases where you are just *adding* things, since it adds those handy `MANAGED BY ANSIBLE` blocks.

Reply:
I have no objections with it. The docs say to just `echo ... >>` so I'll update to use block.